src/Endatix.Infrastructure/Integrations/Slack/SlackClient.cs

```diff
@@ -19,28 +21,64 @@ public class SlackClient : INotificationHandler<SubmissionCompletedEvent>, ISlac
+    private readonly IServiceProvider _serviceProvider;
+    //TODO: The Slack message template should be configurable through the Hub's UI
+    // {0} is the submissionURL, {1} is the form name
+    private const string SLACK_MESSAGE_TEMPLATE = ":page_with_curl: <{0}|New submission> for {1}";
+    // Submission URL parameters: {0} is the Hub's base URL, {1} is FormId
+    private const string SLACK_SUBMISSIONS_URL_TEMPLATE = "{0}/forms/submissions/{1}";
+    public SlackClient(ILogger<SlackClient> logger,
+            IRepository<Form> formRepository,
+            IOptions<SlackSettings> options,
+            IServiceProvider serviceProvider)
+    {
         _formRepository = formRepository;
         _logger = logger;
         _slackSettings = options.Value;
+        _serviceProvider = serviceProvider;
     }
 
-    public async Task Handle(SubmissionCompletedEvent notification, CancellationToken cancellationToken) {
-        if(notification.Submission.IsComplete && _slackSettings.Active){
+    public async Task Handle(SubmissionCompletedEvent notification, CancellationToken cancellationToken)
+    {
+        Guard.Against.Null(_slackSettings.Active, nameof(_slackSettings.Active), "Slack settings must specify whether the integration is active.");
+        Guard.Against.NullOrEmpty(_slackSettings.EndatixHubBaseUrl, nameof(_slackSettings.EndatixHubBaseUrl), "EndatixHubBaseUrl must have a value within the Slack settings collection.");
+        Guard.Against.NullOrEmpty(_slackSettings.Token, nameof(_slackSettings.Token), "Token must have a value within the Slack settings collection.");
+        Guard.Against.NullOrEmpty(_slackSettings.ChannelId, nameof(_slackSettings.ChannelId), "ChannelId must have a value within the Slack settings collection.");
+
+        if (notification.Submission.IsComplete && (bool)_slackSettings.Active)
+        {
+
+            var submissionUrl = string.Format(SLACK_SUBMISSIONS_URL_TEMPLATE, _slackSettings.EndatixHubBaseUrl.TrimEnd('\\', '/'), notification.Submission.FormId);
+            Form? form;
+
+            using (var scope = _serviceProvider.CreateScope())
+            {
+                var repository = scope.ServiceProvider.GetRequiredService<IRepository<Form>>();
+                form = await repository.GetByIdAsync(notification.Submission.FormId, cancellationToken);
+            }
 
-            var submissionUrl = $"{_slackSettings.EndatixHubBaseUrl.TrimEnd('\\', '/')}/forms/submissions/{notification.Submission.FormId}";
-            
-            //TODO: Troubleshoot _formRepository being disposed below and uncomment the line in order to reference the form name correctly
-            // var form = await _formRepository.GetByIdAsync(notification.Submission.FormId, cancellationToken) ?? throw new Exception("Form not found by the FormId of the submission");
+            Guard.Against.Null(form, $"Unable to load submission {notification.Submission.Id}'s Form object.");
```

Comment:
I'm thinking about how this will affect the flow. Putting a guard clause here means leads to an exception if the form is not loaded and a message not published.
Although having the form's name in the message is good, not having the message sent because of missing form's name doesn't seem like the best approach.
I suggest if the form is not loaded a warning is logged so we can investigate and the ID is used in the message. It can be something like:
- "...|New submission> for Contact Us Form" when the form is loaded
- "...|New submission> for form with ID=9328569378562" when the form is not loaded

Reply:
Done

Reply:
It's good that you introduce such fallback logic but it seems it won't really cover the case. It seems `formNameOrId` is set in the case of available form name but is never used. There is no logic that sets the ID in case there is no form and I see a guard clause that fill fail the method execution if there is no form loaded. Please revise.

Reply:
Thanks. Fixed.